tests/framework/extensions/clusters/clusters.go

```diff
@@ -1007,7 +1006,24 @@ func WaitClusterToBeUpgraded(client *rancher.Client, clusterID string) (err erro
 		return
 	}
 
+	return
+}
+
+// WaitClusterUntilUpgrade is a helper function that takes a rancher client, and the cluster id as parameters.
+// Waits until cluster is ready.
+// Cluster error states that declare control plane is inaccessible and cluster object modified are ignored.
+// Same cluster summary information logging is ignored.
+func WaitClusterUntilUpgrade(client *rancher.Client, clusterID string) (err error) {
+	var clusterInfo string
+	opts := metav1.ListOptions{
+		FieldSelector:  "metadata.name=" + clusterID,
+		TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+	}
+
 	watchInterfaceWaitUpgrade, err := client.GetManagementWatchInterface(management.ClusterType, opts)
+	if err != nil {
+		return
+	}
 	checkFuncWaitUpgrade := func(event watch.Event) (bool, error) {
 		clusterUnstructured := event.Object.(*unstructured.Unstructured)
 		summerizedCluster := summary.Summarize(clusterUnstructured)
```

Comment:
nit: "summerized" should be "summarized"

Reply:
Thank you, great catch, addressed